bundles/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java

```diff
@@ -128,6 +128,16 @@ public boolean connect() {
     @Override
     public boolean disconnect() {
         mDevice.setAutoConnect(false);
-        return false;
+        mCmdRun = false;
+        try {
+            mConnectThread.interrupt();
+            mSocket.close();
```

Comment:
You could check that mConnectThread / mSocket are not null.

Reply:
Thanks, added